### ISSUE TITLE
Your decorators don't work as advertised

### ISSUE THREAD

**Issue by kristianmandrup** [croudcare/relpers#8]
First, your relpers `dist` seem to be out of sync with src (and docs).

The npm version, doesn't take the `propertyNames` argument (should be a destructured args anyways)

Also, according to: https://github.com/wycats/javascript-decorators
in order for a decorator to take arguments, it has to be a factory function, taking the arguments and returning the no-args decorator (which can use the args from the enclosing factor function ie. closure scope)

``` js
export function injectProps(...propertyNames) {
  return function (target, name, descriptor) {
```

However, using this approach on my `render` method, I still get an error undefined.

`Uncaught TypeError: Cannot read property 'styles' of null`

``` js
  @injectProps('state', 'props')
  render({ styles }) {
    ...
```

My "fixed" decorator function:

``` js
export function injectProps(...propertyNames) {
  return function (target, name, descriptor) {
    const originalFunction = descriptor.value;

    if (typeof originalFunction !== 'function') {
      throw new SyntaxError(`@injectProps can only be used on functions, not: ${originalFunction}`);
    }

    return {
      ...descriptor,
      value: function propsInjectorWrapper(...args) {
        const properties = propertyNames.map(propertyName => this[propertyName]);
        const newArgs = properties.concat(args);

        return originalFunction.apply(this, newArgs);
      },
    };
  }
}
```

**Comment by kristianmandrup** [comment]
I tried to fix it as follows:

``` js
export function injectProps(...propertyNames) {
  return function (target, name, descriptor) {
    const originalFunction = descriptor.value;

    if (typeof originalFunction !== 'function') {
      throw new SyntaxError(`@injectProps can only be used on functions, not: ${originalFunction}`);
    }

    var fun = {
      ...descriptor,
      value: function propsInjectorWrapper(...args) {
        const properties = propertyNames.reduce((prev, name) => {
          var prop = this[name];
          prop !== null ? prev.push(prop) : prev;
          return prev;
        }, []);

        const newArgs = properties.concat(args);

        return originalFunction.apply(this, newArgs);
      },
    };
    return fun;
  }
}
```

**Comment by goncalvesjoao** [comment]
> First, your relpers dist seem to be out of sync with src (and docs).
> Thanks @kristianmandrup.

When I published the 1.3.2 version I forgot to "npm run build" and the dist code remained locked at the v1.3.1.

Version 1.3.3, now works "as advertised" :+1:
